scipy/signal/spectral.py

```diff
@@ -511,11 +509,401 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+
+    >>> f, t, Zxx = signal.stft(x, fs, nperseg=1000)
+    >>> f = fftshift(f)
+    >>> Zxx = fftshift(Zxx, axes=0)
+    >>> fig, ax = plt.subplots(nrows=2, sharex=True)
+    >>> ax[0].pcolormesh(t, f, np.abs(Zxx))
+    >>> ax[1].pcolormesh(t, f, np.angle(Zxx))
+    >>> ax[0].set_title('STFT Magnitude')
+    >>> ax[0].set_ylabel('Frequency [Hz]')
+    >>> ax[0].set_title('STFT Phase')
+    >>> ax[1].set_ylabel('Frequency [Hz]')
+    >>> ax[1].set_xlabel('Time [sec]')
+    >>> plt.show()
+    """
+
+    freqs, time, Zxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
+                                        nfft=None, detrend=detrend,
+                                        return_onesided=False,
+                                        scaling='spectrum', axis=axis,
+                                        mode='stft', padded=padded,
+                                        centered=centered)
+
+    return freqs, time, Zxx
+
+
+def istft(Zxx, fs=1.0, window='hann', noverlap=128, centered=True,
+          time_axis=-1, freq_axis=-2):
+    """
+    Perform the inverse Short Time Fourier transform (iSTFT).
+
+
+    Parameters
+    ----------
+    Zxx : array_like
+        STFT of the signal to be reconstructed. If a purely real array is
+        passed, it will be cast to a complex data type.
+    fs : float, optional
+        Sampling frequency of the time series. Defaults to 1.0.
+    window : str or tuple or array_like, optional
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length will be used for nperseg.
+        Defaults to a Hann window. Must match the window used to generate
+        the STFT for faithful inversion.
+    noverlap : int, optional
+        Number of points to overlap between segments. If None, half of the
+        segment length. Defaults to None. When specified, the COLA
+        constraint must be met (see Notes below), and should match the
+        parameter used to generate the STFT.
+    centered : bool, optional
+        Specifies whether the input signal was padded via odd extension, by
+        using ``centered=True`` in `stft`. Defaults to True.
+    time_axis : int, optional
+        Where the time segments of the STFT is located; the default is the
+        last axis (i.e. ``axis=-1``).
+    freq_axis : int, optional
+        Where the frequency axis of the STFT is located; the default is the
+        penultimate axis (i.e. ``axis=-2``).
+
+    Returns
+    -------
+    t : ndarray
+        Array of output data times.
+    x : ndarray
+        iSTFT of Zxx.
+
+    See Also
+    --------
+    lombscargle: Lomb-Scargle periodogram for unevenly sampled data
+    welch: Power spectral density by Welch's method.
+    spectrogram: Spectrogram by Welch's method.
+    csd: Cross spectral density by Welch's method.
+    stft: Short Time Fourier Transform
```

Comment:
Same here, `stft` should be listed first. Actually I don't think the others even need to be listed here. It makes sense to have them listed in the `stft` docstring because they do something similar, but they don't do something similar to the `istft`.